app/src/main/java/io/pslab/communication/peripherals/I2C.java

```diff
@@ -140,19 +140,37 @@ public int readStatus() throws IOException {
         return val;
     }
 
-    public ArrayList<Character> readBulk(int deviceAddress, int registerAddress, int bytesToRead) throws IOException {
+    public ArrayList<Integer> readBulk(int deviceAddress, int registerAddress, int bytesToRead) throws IOException {
         packetHandler.sendByte(commandsProto.I2C_HEADER);
         packetHandler.sendByte(commandsProto.I2C_READ_BULK);
         packetHandler.sendByte(deviceAddress);
         packetHandler.sendByte(registerAddress);
         packetHandler.sendByte(bytesToRead);
         byte[] data = new byte[bytesToRead + 1];
         packetHandler.read(data, bytesToRead + 1);
-        ArrayList<Character> charData = new ArrayList<>();
+        ArrayList<Integer> intData = new ArrayList<>();
         for (int i = 0; i < bytesToRead; i++) {
-            charData.add((char) data[i]);
+            intData.add((int) data[i]);
         }
```

Comment:
It should be possible to simplify this loop to:

`for (byte b : data) {`
`   intData.add((int) b);`
`}`

Reply:
Thanks for the suggestion ! I've simplified the loop.